firebase/deploy

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+main() {
+   set -euo pipefail
+
+   if [[ "$#" -gt 1 ]]; then
+       exit_error "usage [CHANNEL=dev]"
+   fi
+
+   channel="dev"
+   if [[ "$#" -eq 1 ]]; then
+       channel="$1"
+   fi
+
+   if [[ -n "${CI:-}" ]]; then
+   	setup_ci_creds
+   fi
+
+   if [[ "${channel}" == "live" ]]; then
+      firebase --config firebase/firebase.json deploy --only hosting
+      exit 0
+   fi
+
+   result=$(firebase --json --config firebase/firebase.json hosting:channel:deploy "${channel}")
+   status=$(jq -r '.status' <<<"${result}")
+
+   if [[ "${status}" == "error" ]]; then
```

Comment:
you could use a `case` statement here if you wish:

```
case "${status}" in
error)
    error_msg=$(jq -r '.error' <<<"${result}")
    exit_error "${error_msg}"
    ;;
success)
    url=$(jq -r '.result."evy-lang".url' <<<"${result}")
    pr_num=$(gh pr view --json number -q .number)
    body="firebase-deployment: $url (${GITHUB_SHA:-$(git rev-parse HEAD)})"
    gh pr comment "${pr_num}" --body "${body}"
    ;;
*)
    exit_error "unknown status: ${result}"
esac
```

That's not very happy-pathy though so I understand if you don't want to - just letting you know the option.

Reply:
hmm, i tried it but I think I prefer what i have, thanks for the insight on the switch though.